safe_transaction_service/history/tasks.py

```diff
@@ -247,13 +248,20 @@ def process_decoded_internal_txs_task(self) -> Optional[int]:
     with contextlib.suppress(LockError):
         with only_one_running_task(self):
             count = 0
+            banned_safes = SafeContract.objects.get_banned_safes()
```

Comment:
Instead of returning a `list`, return a `QuerySet` and cast it here to a `set` so testing inclusion is really fast. Testing inclusion in a set is `O(1)`, testing inclusion in a list is `O(n)`: https://wiki.python.org/moin/TimeComplexity

Reply:
Ok, good suggestion. Then closing this conversation https://github.com/safe-global/safe-transaction-service/pull/1400#discussion_r1158377038